### DISCUSSION TITLE
Reverse Image Search Engine

### DISCUSSION THREAD

**Discussion by batara666** [milvus-io/milvus#4380] [Q&A and General discussion] [answered]
Hi, I'm trying to build reverse image search engine, but i got stuck with numpy.ndarray and postgresql CUBE, will this library help me to build this thing ? or can i just leave the postgresql things ?, thankyou!

**Comment by shiyu22** [comment, marked as answer]
Of course, Milvus was developed precisely for the storage and retrieval of feature vectors. You can easily [insert list of vectors](https://milvus.io/cn/docs/v0.10.4/insert_delete_vector_python.md) in python.

For image search, we also have a related [Bootcamp](https://github.com/milvus-io/bootcamp/blob/0.10.0/solutions/pic_search/README.md) reference.
